Failing tests for v3.3.0 on Alpine Linux

I am building updated package for v3.3.0 and giving followup to all your kind feedback about failing tests.

Now I got following tests failing:

```
=================================== FAILURES ===================================
__________________________ test_selocale_unsupported ___________________________

tmp_path = PosixPath('/tmp/pytest-of-builder/pytest-0/test_selocale_unsupported0')

    def test_selocale_unsupported(tmp_path):
>       with pytest.raises(locale.Error):
E       Failed: DID NOT RAISE <class 'locale.Error'>

tests/i18n/test_i18n.py:27: Failed
__________________ test_ogvjs_installed_script_missing_param ___________________

    def test_ogvjs_installed_script_missing_param():
        # run from installed script to check real conditions
        script = subprocess.run(
            ["/usr/bin/env", "fix_ogvjs_dist"],
            text=True,
            capture_output=True,
            check=False,
        )
>       assert script.returncode == 1
E       assert 127 == 1
E        +  where 127 = CompletedProcess(args=['/usr/bin/env', 'fix_ogvjs_dist'], returncode=127, stdout='', stderr="env: can't execute 'fix_ogvjs_dist': No such file or directory\n").returncode

tests/ogvjs/test_ogvjs.py:55: AssertionError
________________________ test_ogvjs_installed_script_ok ________________________

tmp_path = PosixPath('/tmp/pytest-of-builder/pytest-0/test_ogvjs_installed_script_ok0')
videojs_url = 'https://github.com/videojs/video.js/releases/download/v7.6.4/video-js-7.6.4.zip'
ogvjs_url = 'https://github.com/brion/ogv.js/releases/download/1.6.1/ogvjs-1.6.1.zip'
videojs_ogvjs_url = 'https://github.com/hartman/videojs-ogvjs/archive/v1.3.1.zip'

    @pytest.mark.slow
    def test_ogvjs_installed_script_ok(tmp_path, videojs_url, ogvjs_url, videojs_ogvjs_url):
        # run from installed script to check real conditions
    
        prepare_ogvjs_folder(tmp_path, videojs_url, ogvjs_url, videojs_ogvjs_url)
    
        script = subprocess.run(
            ["/usr/bin/env", "fix_ogvjs_dist", str(tmp_path)],
            text=True,
            capture_output=True,
            check=False,
        )
>       assert script.returncode == 0
E       assert 127 == 0
E        +  where 127 = CompletedProcess(args=['/usr/bin/env', 'fix_ogvjs_dist', '/tmp/pytest-of-builder/pytest-0/test_ogvjs_installed_script_ok0'], returncode=127, stdout='', stderr="env: can't execute 'fix_ogvjs_dist': No such file or directory\n").returncode

tests/ogvjs/test_ogvjs.py:77: AssertionError



...



=========================== short test summary info ============================
FAILED tests/i18n/test_i18n.py::test_selocale_unsupported - Failed: DID NOT RAISE <class 'locale.Error'>
FAILED tests/ogvjs/test_ogvjs.py::test_ogvjs_installed_script_missing_param - assert 127 == 1
FAILED tests/ogvjs/test_ogvjs.py::test_ogvjs_installed_script_ok - assert 127 == 0
=================== 3 failed, 367 passed in 80.17s (0:01:20) ===================

```

The first test as expected fails, so I will continue to ignoring it.

Please, let me know if you need a full check log for ogvjs related errors.

Currently I build the wheel and:

```
python3 -m venv --clear --without-pip --system-site-packages .testenv   
.testenv/bin/python3 -m installer .dist/*.whl
.testenv/bin/python3 -m pytest \
                --runslow \
                --cov=zimscraperlib \
                --cov-report=term --cov-report term-missing
```

--- a/.github/workflows/Tests.yaml
+++ b/.github/workflows/Tests.yaml
@@ -41,6 +41,7 @@ jobs:
         if: matrix.python == '3.12'
         uses: codecov/codecov-action@v3
         with:
+          fail_ci_if_error: true
           token: ${{ secrets.CODECOV_TOKEN }}
 
   build_python:

--- a/.github/workflows/Tests.yaml
+++ b/.github/workflows/Tests.yaml
@@ -35,7 +35,7 @@ jobs:
           pip install -e .[test,scripts]
 
       - name: Run the tests
-        run: inv coverage --args "--runslow -vvv"
+        run: inv coverage --args "--runslow --runinstalled -vvv"
 
       - name: Upload coverage report to codecov
         if: matrix.python == '3.12'

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 - Simplify type annotations by replacing Union and Optional with pipe character ("|") for improved readability and clarity
 
+### Fixed
+- Add back the `--runinstalled` flag for test execution to allow smooth testing on other build chains (#139)
+
 ## [3.3.2] - 2024-03-25
 
 ### Added

--- a/README.md
+++ b/README.md
@@ -55,7 +55,9 @@ apk add ffmpeg gifsicle libmagic wget libjpeg
 
 # Contribution
 
-This project adheres to openZIM's [Contribution Guidelines](https://github.com/openzim/overview/wiki/Contributing)
+This project adheres to openZIM's [Contribution Guidelines](https://github.com/openzim/overview/wiki/Contributing).
+
+This project has implemented openZIM's [Python bootstrap, conventions and policies](https://github.com/openzim/_python-bootstrap/docs/Policy.md) **v1.0.2**.
 
 ```shell
 pip install hatch

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["hatchling", "hatch-openzim"]
+requires = ["hatchling", "hatch-openzim>=0.2"]
 build-backend = "hatchling.build"
 
 [project]
@@ -25,11 +25,10 @@ dynamic = ["authors", "classifiers", "keywords", "license", "version", "urls"]
 
 [tool.hatch.metadata.hooks.openzim-metadata]
 kind = "scraper"
-# not yet supported in hatch-openzim 0.1
-# additional-classifiers = [
-#   "Development Status :: 5 - Production/Stable",
-#   "Intended Audience :: Developers",
-# ]
+additional-classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Intended Audience :: Developers",
+]
 
 [project.optional-dependencies]
 scripts = [

--- a/tasks.py
+++ b/tasks.py
@@ -24,6 +24,7 @@ def report_cov(ctx: Context, *, html: bool = False):
     """report coverage"""
     ctx.run("coverage combine", warn=True, pty=use_pty)
     ctx.run("coverage report --show-missing", pty=use_pty)
+    ctx.run("coverage xml", pty=use_pty)
     if html:
         ctx.run("coverage html", pty=use_pty)
 
@@ -92,7 +93,7 @@ def fix_black(ctx: Context, args: str = "."):
 def fix_ruff(ctx: Context, args: str = "."):
     """fix all ruff rules"""
     args = args or "."  # needed for hatch script
-    ctx.run(f"ruff --fix {args}", pty=use_pty)
+    ctx.run(f"ruff check --fix {args}", pty=use_pty)
 
 
 @task(

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -10,16 +10,28 @@ def pytest_addoption(parser):
     parser.addoption(
         "--runslow", action="store_true", default=False, help="run slow tests"
     )
+    parser.addoption(
+        "--runinstalled",
+        action="store_true",
+        default=False,
+        help="run tests checking for installed features",
+    )
 
 
 def pytest_configure(config):
     config.addinivalue_line("markers", "slow: mark test as slow to run")
+    config.addinivalue_line(
+        "markers", "installed: mark test as testing installed features"
+    )
 
 
 def pytest_collection_modifyitems(config, items):
     skip_slow = pytest.mark.skip(reason="need --runslow option to run")
+    skip_installed = pytest.mark.skip(reason="need --runinstalled option to run")
 
     for item in items:
+        if "installed" in item.keywords and not config.getoption("--runinstalled"):
+            item.add_marker(skip_installed)
         if "slow" in item.keywords and not config.getoption("--runslow"):
             item.add_marker(skip_slow)
 

--- a/tests/ogvjs/test_ogvjs.py
+++ b/tests/ogvjs/test_ogvjs.py
@@ -44,6 +44,7 @@ def prepare_ogvjs_folder(tmp_path, videojs_url, ogvjs_url, videojs_ogvjs_url):
     tmp_path.joinpath(member).rename(tmp_path.joinpath("videojs-ogvjs.js"))
 
 
+@pytest.mark.installed
 def test_ogvjs_installed_script_missing_param():
     # run from installed script to check real conditions
     script = subprocess.run(
@@ -63,6 +64,7 @@ def test_ogvjs_from_code_missing_params():
 
 
 @pytest.mark.slow
+@pytest.mark.installed
 def test_ogvjs_installed_script_ok(tmp_path, videojs_url, ogvjs_url, videojs_ogvjs_url):
     # run from installed script to check real conditions
 

